pkg/controllers/nodeclaim/expiration/controller.go

```diff
@@ -72,9 +78,13 @@ func (c *Controller) Reconcile(ctx context.Context, nodeClaim *v1.NodeClaim) (re
 }
 
 func (c *Controller) Register(_ context.Context, m manager.Manager) error {
-	builder := controllerruntime.NewControllerManagedBy(m)
-	return builder.
+	return controllerruntime.NewControllerManagedBy(m).
 		Named("nodeclaim.expiration").
-		For(&v1.NodeClaim{}).
+		For(&v1.NodeClaim{}, builder.WithPredicates(predicate.TypedFuncs[client.Object]{
```

Comment:
My vote is to keep the same `time.Sleep` mechanism that we have in the NodeClaim lifecycle controller and just apply that to reading our own writes here